tests/nativeapp/test_event_sharing.py

```diff
@@ -0,0 +1,1086 @@
+        install_clause = ""
+
+    calls = [
+        (
+            mock_cursor([("old_role",)], []),
+            mock.call("select current_role()"),
+        ),
+        (None, mock.call("use role app_role")),
+        (
+            mock_cursor([("old_wh",)], []),
+            mock.call("select current_warehouse()"),
+        ),
+        (None, mock.call("use warehouse app_warehouse")),
+        (None, mock.call(f"alter application myapp upgrade {install_clause}")),
+    ]
+
+    if authorize_telemetry_flag is not None:
+        calls.append(
+            (
+                None,
+                mock.call(
+                    f"alter application myapp set AUTHORIZE_TELEMETRY_EVENT_SHARING = {str(authorize_telemetry_flag).upper()}"
+                ),
+            ),
+        )
+
+    if optional_shared_events is not None:
+        calls.append(
+            (
+                None,
+                mock.call(
+                    f"""alter application myapp set shared telemetry events ('{"', '".join(optional_shared_events)}')"""
+                ),
+            ),
+        )
+
+    calls.extend(
+        [
+            (None, mock.call("use warehouse old_wh")),
+            (None, mock.call("use role old_role")),
+        ]
+    )
+    side_effects, mock_execute_query_expected = mock_execute_helper(calls)
+    mock_execute_query.side_effect = side_effects
+    return mock_execute_query_expected
+
+
+@mock.patch(APP_ENTITY_GET_EXISTING_APP_INFO)
+@mock.patch(SQL_EXECUTOR_EXECUTE)
+@mock_connection()
+@mock.patch(
+    GET_UI_PARAMETERS,
+    return_value={
+        UIParameter.EVENT_SHARING_V2: "false",
+        UIParameter.ENFORCE_MANDATORY_FILTERS: "false",
+    },
+)
+@pytest.mark.parametrize(
+    "manifest_contents",
+    [
+        test_manifest_contents,
+        test_manifest_with_mandatory_events,
+    ],
+)
+@pytest.mark.parametrize(
+    "authorize_event_sharing",
+    [
+        False,
+        None,
+    ],
+)
+@pytest.mark.parametrize(
+    "install_method",
+    [
+        SameAccountInstallMethod.unversioned_dev(),
+        SameAccountInstallMethod.release_directive(),
+    ],
+)
+@pytest.mark.parametrize(
+    "is_upgrade",
+    [False, True],
+)
+def test_event_sharing_disabled_no_change_to_current_behavior(
+    mock_param,
+    mock_conn,
+    mock_execute_query,
+    mock_get_existing_app_info,
+    manifest_contents,
+    authorize_event_sharing,
+    install_method,
+    is_upgrade,
+    temp_dir,
+    mock_cursor,
+):
+    mock_execute_query_expected = _setup_mocks_for_app(
+        mock_execute_query,
+        mock_cursor,
+        mock_get_existing_app_info,
+        is_prod=not install_method.is_dev_mode,
+        is_upgrade=is_upgrade,
+    )
+    mock_conn.return_value = MockConnectionCtx()
+    mock_diff_result = DiffResult()
+    _setup_project(
+        manifest_contents=manifest_contents,
+        authorize_event_sharing=authorize_event_sharing,
+    )
+    assert not mock_diff_result.has_changes()
```

Comment:
`mock_diff_result` is never passed anywhere, how can this assertion fail? (also found in other tests)

Reply:
yea it doesn't make sense - i'll remove it from this new file at least.